src/displayapp/screens/Notifications.cpp

```diff
@@ -108,52 +136,82 @@ bool Notifications::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
   }
 
   switch (event) {
+    case Pinetime::Applications::TouchEvents::SwipeRight:
+      if (validDisplay) {
+        Controllers::NotificationManager::Notification previousNotification;
+        auto previousMessage = notificationManager.GetPrevious(currentId);
+        auto nextMessage = notificationManager.GetNext(currentId);
+        if (!previousMessage.valid) {
+          // dismissed last message (like 5/5), need to go one message down (like 4/4)
+          afterDismissNextMessageFromAbove = false; // show next message coming from below
+        } else {
+          afterDismissNextMessageFromAbove = true; // show next message coming from above
+        }
+        notificationManager.Dismiss(currentId);
+        if (previousMessage.valid) {
+          currentId = previousMessage.id;
+        } else if (nextMessage.valid) {
+          currentId = nextMessage.id;
+        } else {
+          // don't update id, won't be found be refresh and try to load latest message or no message box
+        }
+        currentItem.reset(nullptr);
+        app->SetFullRefresh(DisplayApp::FullRefreshDirections::RightAnim);
+        // create black "inTransition" screen
+        currentItem = std::make_unique<NotificationItem>(alertNotificationService, motorController, true);
```

Comment:
I think this can be simplified by just creating a black box that covers the entire screen.

```
lv_obj_t* blackBox = lv_obj_create(lv_scr_act(), nullptr);
lv_obj_set_size(blackBox, LV_HOR_RES, LV_VER_RES);
lv_obj_set_style_local_bg_color(blackBox, LV_OBJ_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
```

Reply:
but wouldn't the black box be created immediately without the screen transition animation?

Reply:
No, because the screen drawing is done by a separate LVGL task, not immediately after object creation. I did a rough test by just replacing this and the transition worked like we want it to.

Reply:
just tried locally in the simulator and it works as you described. Very cool! Less hacky and straightforward solution. Thanks!